File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/javav2/example_code/kinesis/src/main/java/com/example/kinesis/DescribeLimits.java
/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.example.kinesis;
import software.amazon.awssdk.regions.Region;
import software.amazon.awssdk.services.kinesis.KinesisClient;
import software.amazon.awssdk.services.kinesis.model.DescribeLimitsRequest;
import software.amazon.awssdk.services.kinesis.model.DescribeLimitsResponse;
import software.amazon.awssdk.services.kinesis.model.KinesisException;

/**
 * To run this Java V2 code example, ensure that you have setup your development environment, including your credentials.
 *
 * For information, see this documentation topic:
 *
 * https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/get-started.html
 */
public class DescribeLimits {

    public static void main(String[] args) {
        Region region = Region.US_EAST_1;
        KinesisClient kinesisClient = KinesisClient.builder()
                .region(region)
                .build();
        describeKinLimits(kinesisClient);
        kinesisClient.close();
    }
    public static void describeKinLimits(KinesisClient kinesisClient) {
        try {

            DescribeLimitsRequest request = DescribeLimitsRequest.builder()
                 .build();

            DescribeLimitsResponse response = kinesisClient.describeLimits(request);
            System.out.println("Number of open shards: " + response.openShardCount());
            System.out.println("Maximum shards allowed: " + response.shardLimit());

        } catch (KinesisException e) {
            System.err.println(e.getMessage());
            System.exit(1);
        }
        System.out.println("Done");
    }
}

File: docs/static/img/.aws-oxygen/aws-shared/mainline/en_us/code-samples/kotlin/services/kinesis/src/main/kotlin/com/kotlin/kinesis/DescribeLimits.kt
/*
   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
   SPDX-License-Identifier: Apache-2.0
*/

package com.kotlin.kinesis

import aws.sdk.kotlin.services.kinesis.KinesisClient
import aws.sdk.kotlin.services.kinesis.model.DescribeLimitsRequest

/**
To run this Kotlin code example, ensure that you have setup your development environment,
including your credentials.

For information, see this documentation topic:
https://docs.aws.amazon.com/sdk-for-kotlin/latest/developer-guide/setup.html
 */

suspend fun  main(){

    describeKinLimits()
    }

suspend fun describeKinLimits() {

    KinesisClient { region = "us-east-1" }.use { kinesisClient ->
        val response = kinesisClient.describeLimits(DescribeLimitsRequest{})
        println("Number of open shards is ${response.openShardCount}")
        println("Maximum shards allowed is ${response.shardLimit}")

    }
}
